Makefile

```diff
@@ -1,15 +1,66 @@
 -include site.mak
 
+PROGRAMS = pgcollect pginfo pgconvert
 SOURCES = AddressResolver.cpp Profile.cpp
 HEADERS = AddressResolver.h Profile.h
 
-all: pgcollect pginfo pgconvert
+PREFIX = /usr/local
+
+KCACHEGRIND = kcachegrind
+
+CFLAGS = -Wall -Wextra -g
+PGCOLLECT_FLAGS = -F 2048 -s
+
+all: $(PROGRAMS) 
 
 pgcollect: pgcollect.c
-	$(CC) -std=gnu99 -Wall -Wextra -g -D_GNU_SOURCE -o pgcollect pgcollect.c ${FLAGS}
+	$(CC) -std=gnu99  -O2 $(CFLAGS) ${FLAGS} -D_GNU_SOURCE -o pgcollect  pgcollect.c
+
+pgconvert: pgconvert.cpp $(SOURCES) $(HEADERS)
+	$(CXX) -std=c++03 -O2 $(CFLAGS) ${FLAGS}               -o pgconvert  pgconvert.cpp $(SOURCES) -ldw -lelf
 
 pginfo: pginfo.cpp $(SOURCES) $(HEADERS)
-	$(CXX) -std=c++03 -Wall -Wextra -g -o pginfo pginfo.cpp $(SOURCES) -ldw -lelf ${FLAGS}
+	$(CXX) -std=c++03 -O2 $(CFLAGS) ${FLAGS}               -o pginfo     pginfo.cpp    $(SOURCES) -ldw -lelf
 
-pgconvert: pgconvert.cpp $(SOURCES) $(HEADERS)
-	$(CXX) -std=c++03 -Wall -Wextra -g -o pgconvert pgconvert.cpp $(SOURCES) -ldw -lelf ${FLAGS}
+# only used to be traced itself
+pginfo_dbg: pginfo.cpp $(SOURCES) $(HEADERS)
+	$(CXX) -std=c++03 -O  $(CFLAGS) ${FLAGS} -ggdb3        -o pginfo_dbg pginfo.cpp    $(SOURCES) -ldw -lelf
```

Comment:
-ggdb3 could probably indeed help with debugging, but we're not debugging `pginfo_dbg`, but just profiling. Normal `-g` is enough.

Reply:
ok, I've changed the short `-ggdb3` to `-g -fno-omit-frame-pointer` (while `-g` is the default via `CFLAGS` this may be adjusted)